Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +21,111 @@ public class UserCommunicateService : IUserCommunicateService
-
-            if (cityName.ToUpper() == Constants.Commands.Exit)
-            {
-                return false;
-            }
-
-            if (string.IsNullOrEmpty(cityName))
-            {
-                Console.WriteLine(Constants.Validation.EmptyCityName);
-                _logger.LogInformation("The user entered an empty city name. Try again");
-                return true;
-            }
+            Console.WriteLine("Select menu item:");
+            Console.WriteLine("0 - Get currently weather");
+            Console.WriteLine("1 - Get weather for a period of time");
+            Console.WriteLine("2 - Exit");
 
-            var weatherRepresentation = (string)default;
-            try
-            {
-                weatherRepresentation = (await _weatherServiсe.GetByCityNameAsync(cityName)).GetStringRepresentation();
-            }
-            catch (HttpRequestException ex)
+            bool rarseResult = int.TryParse(Console.ReadLine(), out var pointMenu);
+            if (rarseResult && pointMenu >= 0 && pointMenu <= 2)
             {
-                if (ex.StatusCode == HttpStatusCode.NotFound)
+                try
+                {
+                    switch (pointMenu)
+                    {
+                        case 0:
+                            _invoker.SetCommand(new CurrentlyWeatherCommand(this));
+                            await _invoker.RunAsync();
+                            break;
+                        case 1:
+                            _invoker.SetCommand(new ForecastWeatherCommand(this));
+                            await _invoker.RunAsync();
+                            break;
+                        case 2:
+                            _invoker.SetCommand(new ExitCommand());
+                            await _invoker.RunAsync();
+                            break;
+                    }
+                }
+                catch (HttpRequestException ex)
+                {
+                    if (ex.StatusCode == HttpStatusCode.NotFound)
+                    {
+                        Console.WriteLine(Constants.Errors.BadCityName);
+                        _logger.LogError($"Status code: {(int)HttpStatusCode.NotFound} {HttpStatusCode.NotFound}. User entered incorrect city name.");
+                    }
+                    else
+                    {
+                        Console.WriteLine(Constants.Errors.RequestError);
+                        _logger.LogError($"Status code: {(int)ex.StatusCode} {ex.StatusCode}. {ex.Message}");
+                    }
+                }
+                catch (ValidationException ex)
                 {
-                    Console.WriteLine(Constants.Errors.BadCityName);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)HttpStatusCode.NotFound} {HttpStatusCode.NotFound}. User entered incorrect city name.");
-                    return true;
+                    Console.WriteLine(Constants.Validation.IncorrectValues);
```

Comment:
user won't see properties and corresponding error in which he is mistaken?

Reply:
+